### DISCUSSION TITLE
How to create a custom helper?

### DISCUSSION THREAD

**Discussion by melokki** [adonisjs/core#3625] [Help] [answered]
Hello
I am trying to create a custom helper, but I can't figure out how to do it
I've created a new folder `helpers` where I've placed my code
But when I run my code to helper is not available and I get the following error

`Cannot find module 'helpers/listToTree'`

in `.adonisrc.json` file at `preloads` key I have the following

```
"preloads": [
    "./start/routes",
    "./start/kernel",
    "./helpers/listToTree"
  ],
```

Any suggestion?

**Comment by RomainLanz** [comment]
Hey @melokki! 👋🏻 

It seems like an [XY Problem](https://xyproblem.info/).
What are you trying to achieve (removing the fact that loading your file is not working)?

**Reply by melokki** [reply]
Hello, thank you for your reply :)

I am trying to create a helper function that transforms a flat array into a tree. I might need this function in other parts of the application and I thought that making this function as a helper would be a good idea.

**Reply by RomainLanz** [reply]
Yeah, but why are you trying to preload the file?

Create a simple JS file anywhere you want (like in `App/Helpers`), and import it like in any JS codebase.

```js
import myHelpers from 'App/Helpers'

// ...
```

**Reply by melokki** [reply]
I tried that and it didn't work, I got the same error and I thought maybe I need to preload it or something. Will give it a try again maybe I did something wrong. thank you very much for your time :)

**Comment by melokki** [comment, marked as answer]
I managed to make it working after I did the following:

In the `root` folder of the application I've created a new folder called `helpers`

in `.adonisrc.json` file I added this line:

```ts
"aliases": {
   ...
    "Helpers": "helpers"
  },
```

and in `tsconfig.json` I've added the following:

```ts
"compilerOptions": {
    "outDir": "build",
    "rootDir": "./",
    "sourceMap": true,
    "paths": {
      ...
      "Helpers/*": [
        "./helpers/*"
      ]
    },
```

Now I can make the import from `Helpers` and the code works just fine.
I don't know if it's the right way

**Reply by AbidKhairyAK** [reply]
for adonisjs v6, instead of writing alias in .adonisrc.json file, you need to write subpath alias in your package.json

**Comment by backend-sempoa** [comment]
can you guys implement this in command file? @melokki

**Comment by gregmsanderson** [comment]
Yep, a `node ace make:helper name`.

I was looking how to add a nanoid. Closest is https://docs.adonisjs.com/guides/references/helpers#cuid but it could be nice to add to those provided helpers in an "offficial" way
